go/service/chat_local.go

```diff
@@ -126,251 +207,141 @@ func (h *chatLocalHandler) GetThreadLocal(ctx context.Context, arg chat1.GetThre
+				MessageType: chat1.MessageType_METADATA,
+				Prev:        nil, // TODO
+				// Sender and SenderDevice filled by prepareMessageForRemote
+			},
+			MessageBody: chat1.NewMessageBodyWithMetadata(
+				chat1.MessageConversationMetadata{
+					ConversationTitle: *topicName,
+				}),
 		}
-		return chat1.ResolveConversationLocalRes{
-			Convs:      []chat1.ConversationInfoLocal{info},
-			RateLimits: h.aggRateLimits(rlimits),
-		}, nil
-	}
-	rcres, err := h.resolveConversations(ctx, arg, &rlimits)
-	if err != nil {
-		return chat1.ResolveConversationLocalRes{}, err
-	}
-	return chat1.ResolveConversationLocalRes{
-		Convs:      rcres,
-		RateLimits: h.aggRateLimits(rlimits),
-	}, nil
-}
-
-func (h *chatLocalHandler) resolveConversationsPipeline(ctx context.Context, convs []chat1.Conversation) ([]chat1.ConversationLocal, error) {
-	// Fetch conversation local information in parallel
-	ctx, _ = getUserInfoMapper(ctx, h.G())
-	type jobRes struct {
-		conv  chat1.ConversationLocal
-		index int
-	}
-	type job struct {
-		conv  chat1.Conversation
-		index int
-	}
-	eg, ctx := errgroup.WithContext(ctx)
-	convCh := make(chan job)
-	retCh := make(chan jobRes)
-	eg.Go(func() error {
-		defer close(convCh)
-		for i, conv := range convs {
-			select {
-			case convCh <- job{conv: conv, index: i}:
-			case <-ctx.Done():
-				return ctx.Err()
-			}
+	} else {
+		v1 = chat1.MessagePlaintextV1{
+			ClientHeader: chat1.MessageClientHeader{
+				Conv:        triple,
+				TlfName:     tlfName,
+				TlfPublic:   tlfVisibility == chat1.TLFVisibility_PUBLIC,
+				MessageType: chat1.MessageType_TLFNAME,
+				Prev:        nil, // TODO
+				// Sender and SenderDevice filled by prepareMessageForRemote
+			},
 		}
-		return nil
-	})
-	for i := 0; i < 10; i++ {
-		eg.Go(func() error {
-			for conv := range convCh {
-				info, maxMessages, err := h.getConversationInfo(ctx, conv.conv)
-				jr := jobRes{
-					conv: chat1.ConversationLocal{
-						Info:     &info,
-						Messages: maxMessages,
-						ReadUpTo: conv.conv.ReaderInfo.ReadMsgid,
-					},
-					index: conv.index,
-				}
-				if err != nil {
-					jr.conv.Error = new(string)
-					*jr.conv.Error = err.Error()
-				}
-				select {
-				case retCh <- jr:
-				case <-ctx.Done():
-					return ctx.Err()
-				}
-			}
-			return nil
-		})
-	}
-	go func() {
-		eg.Wait()
-		close(retCh)
-	}()
-	res := make([]chat1.ConversationLocal, len(convs))
-	for c := range retCh {
-		res[c.index] = c.conv
 	}
-	if err := eg.Wait(); err != nil {
-		return nil, err
-	}
-	return res, nil
+	return h.prepareMessageForRemote(ctx, chat1.NewMessagePlaintextWithV1(v1))
 }
 
-func (h *chatLocalHandler) GetInboxSummaryLocal(ctx context.Context, arg chat1.GetInboxSummaryLocalQuery) (res chat1.GetInboxSummaryLocalRes, err error) {
+func (h *chatLocalHandler) GetInboxSummaryForCLILocal(ctx context.Context, arg chat1.GetInboxSummaryForCLILocalQuery) (res chat1.GetInboxSummaryForCLILocalRes, err error) {
 	if err = h.assertLoggedIn(ctx); err != nil {
```

Comment:
Will this function ever have a user for the GUI? I'm wondering if the name is too specific.